shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_FlutterAccessibilityMac_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_FlutterAccessibilityMac_H_
+
+#import <Cocoa/Cocoa.h>
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterEngine.h"
+
+#import "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/third_party/accessibility/flutter_accessibility.h"
+
+#define kRootNode 0
+
+namespace ax {
+
+class FlutterAccessibilityMac : public FlutterAccessibility {
+ public:
+  FlutterAccessibilityMac();
+  ~FlutterAccessibilityMac();
+  // FlutterAccessibility override
+  void Init(AccessibilityBridge* bridge, AXNode* node) override;
+  void OnAccessibilityEvent(AXEventGenerator::TargetedEvent targeted_event) override;
+  void DispatchAccessibilityAction(uint16_t target,
+                                   FlutterSemanticsAction action,
+                                   uint8_t* data,
+                                   size_t data_size) override;
+  gfx::NativeViewAccessible GetNativeViewAccessible() override;
+  gfx::NativeViewAccessible GetParent() override;
+  SkRect GetBoundsRect(const AXCoordinateSystem coordinate_system,
+                       const AXClippingBehavior clipping_behavior,
+                       AXOffscreenResult* offscreen_result) const override;
+  gfx::NativeViewAccessible GetNSWindow() override;
+
+ private:
```

Comment:
General question: can we use smart pointers instead of raw pointers for any/all of these?

i.e. std::shared_ptr or std::unique_ptr